### ISSUE TITLE
Support for QR, Cholesky and Linear Solve

### ISSUE THREAD

**Issue by mandar2812** [eaplatanios/tensorflow_scala#84]
The tensorflow API current supports QR, Cholesky and solving of Linear systems, it would be nice if there was a handle for them in the scala API as well.

**Comment by eaplatanios** [comment]
@mandar2812 Yeah it would indeed be good to support those. My API currently allows you to easily add support for them. For example, you can see how math ops are supported [here](https://github.com/eaplatanios/tensorflow_scala/blob/master/api/src/main/scala/org/platanios/tensorflow/api/ops/Math.scala) and the equivalent Python code is [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/math_ops.py). Similar you could implement support for the linear algebra ops by porting [this Python module](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/linalg_ops.py) (gradients implementation is [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/linalg_grad.py)). It would be great to have that, but unfortunately I don't have time to implement this at the moment. I'm working on a machine translation library that I'll be releasing soon. Pull requests are always appreciated though. :)

I also wrote some brief instructions in #32 a while ago, that might be helpful.
